ruma-events/src/key/verification.rs

```diff
@@ -30,6 +30,8 @@ pub enum HashAlgorithm {
 pub enum KeyAgreementProtocol {
     /// The [Curve25519](https://cr.yp.to/ecdh.html) key agreement protocol.
     Curve25519,
+    /// The Curve25519 key agreement protocol with check for public keys.
+    Curve25519HkdfSha256,
```

Comment:
Can you add a test that verifies that this (de)serializes to the expected string? I think the `snake_case` conversion will do the right thing here, but I'm not entirely certain.

Reply:
Oh wait, it needs kebab case, no? The actual string is `curve25519-hkdf-sha256`.

Reply:
Oh, right! Good thing we're talking about this!